Finished methods from equipment facade in equipmentController

DenisFigula opened the pull request:
closes #115

DenisFigula committed:
Finished methods from equipment facade in equipmentController
closes #115

DenisFigula committed:
Finished methods from equipment facade in equipmentController
closes #115

melanger committed:
enhancements

melanger committed:
enhancements

melanger committed:
plural

melanger committed:
plural

melanger reviewed (changes requested):
I guess we should have sketched the API, my fault :) Please see my changes, I would like it more this way. I recommend http://www.restapitutorial.com/lessons/restquicktips.html or similar reading.

- considering the `by_member_id` path, I would ideally put it as `/members/{member_id}/equipment` (in `MemberController`)
- when I examined the `EquipmentFacade.getEquipmentByMember` method more carefully, I think it can never return `null`, because `equipmentService` delegates to `equipmentDao` which calls [getResultList](https://docs.oracle.com/javaee/7/api/javax/persistence/TypedQuery.html#getResultList--), which returns an empty list in case there no matches and uses exceptions for errors (that's something I missed in the facade)
- on the other hand, please check what happens if memberId is not in the database (I created a test for that in `EquipmentFacadeTest`), it might be a good idea to alter it so that it really returns `null` in this case and then the REST API method is correct (or throw an exception and re-throw in the REST layer)
- I think `POST` is more accurate for creation if the id is autogenerated (you can surely argument on this one and change it back)
- `getEquipment` (`/{id}`) and `addEquipmentToMember` (`/{member_id}`) had the same URL mapping ("slash number")

Also, please try to write either tests for the REST controller, or (as is needed anyway) write some cURL commands which can be used to test that the REST API works. You can create a file called REST.md in the root of the repository and place them there (#108)

DenisFigula committed:
Implemented TourController, some improvements
test process in README.md will be added in different issue
closes #118

DenisFigula committed:
Implemented TourController, some improvements
test process in README.md will be added in different issue
closes #118

melanger committed:
PhotoEntity interface (needs merging with tourController branch)

melanger committed:
improvements - catching specific exceptions instead of all

melanger committed:
improvements - catching specific exceptions instead of all

melanger commented:
I am not quite sure if you did that by mistake or on purpose. Anyway, as I already said:

- [ ] write a cURL command (or a test) that uses `getEquipmentByMember` and the response is 404 (that assumes an exception is thrown by `getEquipmentByMember`, which I don't think is possible). I think you should have left my version with `if (equipmentDTOList == null)` together with your change of `EquipmentFacade.getEquipmentByMember`
- [ ] if you're adding something, how can it be "not found"? In `addEquipmentToMember` the member could have been not found, but I think it's more appropriate to use our `InvalidParameterException` (HTTP not acceptable). Also I modified `EquipmentFacadeImpl.addEquipmentToMember` to throw `IllegalArgumentException` instead of implicitly throwing `NullPointerException`

For the deletion, I introduced a custom exception (instead of the implicit `NullPointerException` that your were catching with `catch (Exception e)`).

Please, don't catch `Exception` or tell me what is your reason behind that. If needed, you can edit facades, services etc. to throw an appropriate exception and catch only that. (For example, you were reacting to a `DataAccessException` which can mean for example that the database is temporarily down with a 404 Not Found)

DenisFigula committed:
Implemented TourController, some improvements
test process in README.md will be added in different issue
closes #118

DenisFigula committed:
getEquipmentByMember method adjusted

melanger committed:
merge master

melanger committed:
getEquipmentByMember method adjusted fix

melanger committed:
exception instead of null in EquipmentFacade.get

melanger committed:
renamed (Photo)EntityNotFoundException

melanger committed:
memberId moved to EquipmentCreateDTO

+ exception unification
+ fixed add/removeEquipment not calling service.add/remove

melanger commented:
Now @DenisFigula please write the cURL commands, test the endpoints and fix bugs if needed.
